numpy/fft/_helper.py

```diff
@@ -163,19 +163,24 @@ def fftfreq(n, d=1.0, device=None):
     >>> freq = np.fft.fftfreq(n, d=timestep)
     >>> freq
     array([ 0.  ,  1.25,  2.5 , ..., -3.75, -2.5 , -1.25])
-
     """
-    if not isinstance(n, integer_types):
-        raise ValueError("n should be an integer")
+    try:
+        n = operator.index(n)
+    except TypeError:
+        raise TypeError("n must be an integer")
```

Comment:
TBH, I would just keep the original error, but I suppose Ok either way.  The original error gives information about what `n` was, the only downside is that it doesn't mention `n` (but there is only one parameter anyway).